docs/nx/next/infra.md

```diff
@@ -0,0 +1,173 @@
+### @ensono-stacks/next:infrastructure
+
+<details>
+<summary>Configure Infrastructure for your Next project</summary>
+
+The infrastructure generator will provide all the necessary tools and setup ready to configure your pipeline.
+
+## Prerequisites
+
+An existing [Next](https://nextjs.org/) application. This may already exist if you agreed to install the infra during next:init generator.
+
+## Usage
+
+```bash
+nx g @ensono-stacks/next:infrastructure
+```
+
+### Command line arguments
+
+The following command line arguments are available:
+
+| Option    | Description             | Type              | Accepted Values | Default |
+| --------- | ----------------------- | ----------------- | --------------- | ------- |
+| --project | The name of the project | nameOfApplication | string          | N/A     |
+
+### Generator Output
+
+```
+├── workspace root
+    ├── apps
+        ├── myapp
+            ├── build
+                ├── helm
+                ├── terraform
+```
+
+##### The next auth plugin will
+
+- Create numerous files under the two folders, helm and terraform. You can then go in and update relevant parts for your use case.
+
+- Adds following files to .gitignore
+
+```
+'**/.terraform/*',
+'*.tfstate',
+'*.tfstate.*',
+'crash.log',
+'crash.*.log',
+'override.tf',
+'override.tf.json',
+'*_override.tf',
+'*_override.tf.json',
+'.terraformrc',
+'terraform.rc',
+```
+
+- installs following dev dependencies
+
+```
+@nx-tools/nx-container
+@nx-tools/container-metadata
+@jscutlery/semver
+```
+
+- Adds `stacks` object to nx.json which is populated via interactive and pulled in various stages of the infra journey
```

Comment:
It shouldn't add anything realted to nx.json. but reads it to know how to scaffold the infra as code values. Its a requirement

Reply:
I had this on another plugin - I believe the first generator you run after `create-stacks-workspace` may add some fields to nx.json depending on what's already there - worth checking if this file always gets changed depending on settings etc